controllers/applications.ts

```diff
@@ -0,0 +1,147 @@
+import { addLog } from "../models/logs";
+const { logType } = require("../constants/logs");
+import { CustomRequest, CustomResponse } from "../types/global";
+const { INTERNAL_SERVER_ERROR } = require("../constants/errorMessages");
+const ApplicationModel = require("../models/applications");
+const { API_RESPONSE_MESSAGES } = require("../constants/application");
+const { getUserApplicationObject } = require("../utils/application");
+const admin = require("firebase-admin");
+
+const getAllOrUserApplication = async (req: CustomRequest, res: CustomResponse): Promise<any> => {
+  try {
+    const { userId, status, next, size } = req.query;
+    const limit = Number(size) || 25;
+    let nextPageUrl = null;
+
+    if (userId) {
+      const applications = await ApplicationModel.getUserApplications(userId);
+
+      return res.json({
+        message: "User applications returned successfully!",
+        applications,
+      });
+    }
+
+    if (status) {
+      const { applications, lastDocId } = await ApplicationModel.getApplicationsBasedOnStatus(status, limit, next);
+
+      if (applications.length === limit) {
+        nextPageUrl = `/applications?next=${lastDocId}&size=${limit}&status=${status}`;
+      }
+
+      return res.json({
+        message: API_RESPONSE_MESSAGES.APPLICATION_RETURN_SUCCESS,
+        applications,
+        next: nextPageUrl,
+      });
+    }
+
+    const { applications, lastDocId } = await ApplicationModel.getAllApplications(limit, next);
+
+    if (applications.length === limit) {
+      nextPageUrl = `/applications?next=${lastDocId}&size=${limit}`;
+    }
+
+    return res.json({
+      message: API_RESPONSE_MESSAGES.APPLICATION_RETURN_SUCCESS,
+      applications,
+      next: nextPageUrl,
+    });
+  } catch (err) {
+    logger.error(`Error in fetching application: ${err}`);
+    return res.boom.badImplementation(INTERNAL_SERVER_ERROR);
+  }
+};
+
+const addApplication = async (req: CustomRequest, res: CustomResponse) => {
+  try {
+    const rawData = req.body;
+    const { applications } = await ApplicationModel.getApplicationsBasedOnStatus("pending", 1, "", req.userData.id);
+    if (applications.length) {
+      return res.status(409).json({
+        message: "User application is already present!",
+      });
+    }
+    const createdAt = admin.firestore.Timestamp.fromDate(new Date());
```

Comment:
Let's use ISO time

Reply:
When I do this new Date().toISOString(), firebase start to ignore it, because it either want a firestore timestamp, if we are order by that field